### PR TITLE
VIH-7094 JVS Endpoint

### PR DESCRIPTION
**Pull request by levibutler**
### JIRA link (if applicable) ###
VIH-7094

### Change description ###
Support JVS endpoints in private consultations

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/services/api/consultation.service.ts [hunk 1]
```diff
@@ -94,24 +93,19 @@ export class ConsultationService {
...
-     * Start a private consultation with video endpoint. This will only be allowed for defence advocates linked to the
-     * endpoint
-     * @param conference conference
-     * @param endpoint video endpoint to call
-     */
-    async startPrivateConsulationWithEndpoint(conference: ConferenceResponse, endpoint: VideoEndpointResponse) {
-        this.logger.info(`[ConsultationService] - Starting a private consultation with a video endpoint`, {
-            conference: conference.id,
-            endpoint: endpoint.id
+    async inviteToConsulation(conferenceId: string, roomLabel: string, requestParticipantId: string) {
```
Comment (surfsquid) [inline VideoWeb/VideoWeb/ClientApp/src/app/services/api/consultation.service.ts:96]: typo: inviteToConsulation

File: VideoWeb/VideoWeb/ClientApp/src/app/services/api/consultation.service.ts [hunk 2]
```diff
@@ -121,18 +115,18 @@ export class ConsultationService {
         }
     }
 
-    async inviteToConsulation(conferenceId: string, roomLabel: string, requestParticipantId: string) {
-        this.logger.info(`[ConsultationService] - Inviting participant to this private consultation`, {
+    async addEndpointToConsulation(conferenceId: string, roomLabel: string, endpointId: string) {
```
Comment (surfsquid) [inline VideoWeb/VideoWeb/ClientApp/src/app/services/api/consultation.service.ts:118]: typo: Consulation

File: VideoWeb/VideoWeb/ClientApp/src/app/testing/mocks/mock-consultation-service.ts [hunk 3]
```diff
@@ -15,7 +15,8 @@ export function consultationServiceSpyFactory(): jasmine.SpyObj<ConsultationServ
         'inviteToConsulation',
         'lockConsultation',
         'displayConsultationLeaveModal',
-        'joinPrivateConsultationRoom'
+        'joinPrivateConsultationRoom',
+        'addEndpointToConsulation'
```
Comment (surfsquid) [inline VideoWeb/VideoWeb/ClientApp/src/app/testing/mocks/mock-consultation-service.ts:19]: typo: Consulation

File: VideoWeb/VideoWeb/ClientApp/src/app/services/api/consultation.service.ts (hunk 2 above)
Comment (levibutler) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/api/consultation.service.ts:118]: fixed

File: VideoWeb/VideoWeb/ClientApp/src/app/services/api/consultation.service.ts (hunk 1 above)
Comment (levibutler) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/api/consultation.service.ts:96]: fixed

File: VideoWeb/VideoWeb/ClientApp/src/app/testing/mocks/mock-consultation-service.ts (hunk 3 above)
Comment (levibutler) [reply VideoWeb/VideoWeb/ClientApp/src/app/testing/mocks/mock-consultation-service.ts:19]: fixed

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts [hunk 4]
```diff
@@ -264,16 +275,22 @@ export abstract class WaitingRoomBaseComponent {
...
+                if (participant) {
+                    participant.current_room = null;
+                    if (roomTransfer.to_room.toLowerCase().indexOf('consultation') >= 0) {
+                        const room = this.conferenceRooms.find(r => r.label === roomTransfer.to_room);
+                        participant.current_room = room
+                            ? new RoomSummaryResponse(room)
+                            : new RoomSummaryResponse({ label: roomTransfer.to_room });
+                    }
+                } else {
+                    const endpoint = this.conference.endpoints.find(p => p.id === roomTransfer.participant_id);
```
Comment (levibutler) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts:287]: could move to top and use else if to handle it not being either? - VHO?

**Review by surfsquid — DISMISSED** [review]

**Review by valentinabennett — APPROVED** [review]

**Pull request merged**